src/core/net/netif.cpp

```diff
@@ -336,6 +336,9 @@ otError Netif::SubscribeExternalMulticast(const Address &aAddress)
     VerifyOrExit(entry != nullptr, error = OT_ERROR_NO_BUFS);
 
     entry->mAddress = aAddress;
+#if (OPENTHREAD_FTD || OPENTHREAD_MTD) && OPENTHREAD_CONFIG_MLR_ENABLE
```

Comment:
is `(OPENTHREAD_FTD || OPENTHREAD_MTD)` necessary?

Reply:
removed